dace/codegen/targets/cpu.py

```diff
@@ -1139,24 +1140,28 @@ def memlet_definition(self,
         memlet_type = conntype.dtype.ctype
 
         desc = sdfg.arrays[memlet.data]
-        ptr = cpp.ptr(memlet.data, desc)
 
         var_type, ctypedef = self._dispatcher.defined_vars.get(memlet.data)
+
+        try:
+            ptr = cpp.ptr(
+                memlet.data, desc, memlet.subset, sdfg, output,
+                self._dispatcher, 0, var_type == DefinedType.ArrayInterface
+                and not isinstance(desc, data.View))
+        except ValueError:
+            raise cgx.CodegenError(
+                "Only memlets accessing a single HBM-bank may be "
+                f"attached to a tasklet. See {str(memlet)}")
```

Comment:
In ptr we dont know that it is because of a tasklet. But really this is verified in the validation, so I guess it should be save to just remove this seperate check

Reply:
Let's remove it here, then. My problem was with `except ValueError`, as this is too broad in case something else throws a `ValueError`!